src/rosdistro/dependency_walker.py

```diff
@@ -127,6 +130,8 @@ def _get_dependencies(self, pkg_name, dep_type):
             'test': pkg.test_depends,
             'doc': pkg.doc_depends,
         }
+        if self._evaluate_condition_context:
+            return set([d.name for d in deps[dep_type] if d.evaluated_condition])
```

Comment:
To avoid the condition you could always check for `if d.evaluated_condition is not False`.

Reply:
Done!